lib/ansible/vars/manager.py

```diff
@@ -575,6 +580,7 @@ def set_nonpersistent_facts(self, host, facts):
         Sets or updates the given facts for a host in the fact cache.
         """
 
+        warn_if_reserved(facts.keys())
```

Comment:
This should go below the next two lines in case facts isn't a dict.

Reply:
rright!